File: evals/fixtures/express-ts/app/src/routes/reservations.ts
import { Router, Request, Response } from "express";
import {
  createReservation,
  updateReservation,
  cancelReservation,
  getReservation,
} from "../services/reservationService";

const router = Router();

router.post("/", (req: Request, res: Response) => {
  try {
    const { guestName, guestEmail, unitId, checkIn, checkOut } = req.body;

    if (!guestName || !guestEmail || !unitId || !checkIn || !checkOut) {
      res.status(400).json({
        error: "Missing required fields: guestName, guestEmail, unitId, checkIn, checkOut",
      });
      return;
    }

    const reservation = createReservation({
      guestName,
      guestEmail,
      unitId,
      checkIn,
      checkOut,
    });

    res.status(201).json({ reservation });
  } catch (err: any) {
    if (err.message?.includes("not found")) {
      res.status(404).json({ error: err.message });
    } else {
      res.status(400).json({ error: err.message });
    }
  }
});

router.put("/:id", (req: Request, res: Response) => {
  try {
    const { checkIn, checkOut } = req.body;
    const reservation = updateReservation(req.params.id, { checkIn, checkOut });
    res.status(200).json({ reservation });
  } catch (err: any) {
    if (err.message?.includes("not found")) {
      res.status(404).json({ error: err.message });
    } else {
      res.status(400).json({ error: err.message });
    }
  }
});

router.delete("/:id", (req: Request, res: Response) => {
  try {
    const reservation = cancelReservation(req.params.id);
    res.status(200).json({ reservation });
  } catch (err: any) {
    if (err.message?.includes("not found")) {
      res.status(404).json({ error: err.message });
    } else {
      res.status(400).json({ error: err.message });
    }
  }
});

router.get("/:id", (req: Request, res: Response) => {
  try {
    const reservation = getReservation(req.params.id);
    res.status(200).json({ reservation });
  } catch (err: any) {
    if (err.message?.includes("not found")) {
      res.status(404).json({ error: err.message });
    } else {
      res.status(400).json({ error: err.message });
    }
  }
});

export default router;

File: evals/fixtures/flask-py/app/blueprints/reservations.py
from flask import Blueprint, jsonify, request

from services.reservation_service import (
    cancel_reservation,
    create_reservation,
    get_reservation,
    update_reservation,
)

reservations_bp = Blueprint("reservations", __name__, url_prefix="/api/reservations")


@reservations_bp.route("", methods=["POST"])
def create():
    try:
        reservation = create_reservation(request.json)
        return jsonify({"reservation": reservation}), 201
    except ValueError as e:
        return jsonify({"error": str(e)}), 400


@reservations_bp.route("/<reservation_id>", methods=["PUT"])
def update(reservation_id):
    try:
        reservation = update_reservation(reservation_id, request.json)
        return jsonify({"reservation": reservation}), 200
    except LookupError as e:
        return jsonify({"error": str(e)}), 404


@reservations_bp.route("/<reservation_id>", methods=["DELETE"])
def cancel(reservation_id):
    try:
        reservation = cancel_reservation(reservation_id)
        return jsonify({"reservation": reservation}), 200
    except LookupError as e:
        return jsonify({"error": str(e)}), 404


@reservations_bp.route("/<reservation_id>", methods=["GET"])
def get(reservation_id):
    try:
        reservation = get_reservation(reservation_id)
        return jsonify({"reservation": reservation}), 200
    except LookupError as e:
        return jsonify({"error": str(e)}), 404
